src/aggregate/expr/expression.c

```diff
@@ -144,8 +144,11 @@ static int evalInverted(ExprEval *eval, const RSInverted *vv, RSValue *result) {
 static int evalPredicate(ExprEval *eval, const RSPredicate *pred, RSValue *result) {
   RSValue l = RSVALUE_STATIC, r = RSVALUE_STATIC;
   int rc = EXPR_EVAL_ERR;
-  if (evalInternal(eval, pred->left, &l) == EXPR_EVAL_ERR ||
-      evalInternal(eval, pred->right, &r) == EXPR_EVAL_ERR) {
+  if (evalInternal(eval, pred->left, &l) == EXPR_EVAL_ERR) {
+    goto cleanup;
+  } else if (pred->cond == RSCondition_Or && RSValue_BoolTest(&l)) {
```

Comment:
@ashtul what if the condition is 'and' and left side is false? there is also not need to calculate the right side, I think the better approach is that the OR/AND will call evaInternal (evaInternal(right) && evaInternal(left) for example), this way the runtime will decide if calculation of the right size is needed.

Reply:
@MeirShpilraien 
Is it OK like this?

Reply:
I like Meir's approach in theory, but the problem is finding a good way to report an error if and when it happens. It's still possible but woud require another helper function (or macro) and would perhaps complicate readability more than it should.

This approach would work fine for `&&` where there is only one possible success outcome `(true && true)`, but for `_Or` we need to be careful to not allow something like `if (x() || y())` where `x()` returns an error.

Reply:
This is not what I meant, I meant something like:
if(pred->cond == RSCondition_And){
   evan(left, &err) && eval(right, &err)
}else (pred->cond == RSCondition_Or){
   evan(left, &err) || eval(right, &err)
}

And as Mark said, error need to be handled differently, maybe a return value..